Sources/X509/Verifier/VerificationDiagnostic.swift

```diff
@@ -0,0 +1,237 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCertificates open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftCertificates project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftCertificates project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import SwiftASN1
+
+public struct VerificationDiagnostic: Sendable {
```

Comment:
Yes, I don't think there is currently a reason why they would want to. We do not guarantee what diagnostics we produce and don't want to. `Verifier` wants to find a valid chain as fast as possible and doesn't check all possible paths as it would be rather expensive (and in theory impossible because it can contains loops). This means that the diagnostics can and will change if we improve the graph traversal algorithm or other parts of the verifier. `VerificationDiagnostic` is also not `Equatable` for the same reason.